sdk/core/azure-core/src/main/java/com/azure/core/http/rest/SyncRestProxy.java

```diff
@@ -0,0 +1,692 @@
+ */
+public final class SyncRestProxy implements InvocationHandler {
+    private static final ByteBuffer VALIDATION_BUFFER = ByteBuffer.allocate(0);
+    private static final String BODY_TOO_LARGE = "Request body emitted %d bytes, more than the expected %d bytes.";
+    private static final String BODY_TOO_SMALL = "Request body emitted %d bytes, less than the expected %d bytes.";
+    private static final String MUST_IMPLEMENT_PAGE_ERROR =
+        "Unable to create PagedResponse<T>. Body must be of a type that implements: " + Page.class;
+
+    private static final ResponseConstructorsCache RESPONSE_CONSTRUCTORS_CACHE = new ResponseConstructorsCache();
+
+    // RestProxy is a commonly used class, use a static logger.
+    private static final ClientLogger LOGGER = new ClientLogger(SyncRestProxy.class);
+
+    private final HttpPipeline httpPipeline;
+    private final SerializerAdapter serializer;
+    private final SwaggerInterfaceParser interfaceParser;
+    private final HttpResponseDecoder decoder;
+
+    /**
+     * Create a RestProxy.
+     *
+     * @param httpPipeline the HttpPipelinePolicy and HttpClient httpPipeline that will be used to send HTTP requests.
+     * @param serializer the serializer that will be used to convert response bodies to POJOs.
+     * @param interfaceParser the parser that contains information about the interface describing REST API methods that
+     * this RestProxy "implements".
+     */
+    private SyncRestProxy(HttpPipeline httpPipeline, SerializerAdapter serializer, SwaggerInterfaceParser interfaceParser) {
+        this.httpPipeline = httpPipeline;
+        this.serializer = serializer;
+        this.interfaceParser = interfaceParser;
+        this.decoder = new HttpResponseDecoder(this.serializer);
+    }
+
+    /**
+     * Get the SwaggerMethodParser for the provided method. The Method must exist on the Swagger interface that this
+     * RestProxy was created to "implement".
+     *
+     * @param method the method to get a SwaggerMethodParser for
+     * @return the SwaggerMethodParser for the provided method
+     */
+    private SwaggerMethodParser getMethodParser(Method method) {
+        return interfaceParser.getMethodParser(method);
+    }
+
+    /**
+     * Send the provided request asynchronously, applying any request policies provided to the HttpClient instance.
+     *
+     * @param request the HTTP request to send
+     * @param contextData the context
+     * @return a {@link Mono} that emits HttpResponse asynchronously
+     */
+    public HttpResponse send(HttpRequest request, Context contextData) {
+        return httpPipeline.sendSynchronously(request, contextData);
+    }
+
+    @Override
+    public Object invoke(Object proxy, final Method method, Object[] args) {
+        validateResumeOperationIsNotPresent(method);
+
+        try {
+            final SwaggerMethodParser methodParser = getMethodParser(method);
```

Comment:
I wonder if a better option is to add metadata that tracks if the method called was synchronous or asynchronous into SwaggerMethodParser. This may also help simplify future work with regards of client telemetry that is sent including information such as whether the service client method called was sync or async. Additionally, this may allow us to continue using a single InvocationHandler implementation (RestProxy) instead of having a split path for sync vs async